WireDomain/Sources/WireDomain/Notifications/GenerateNotificationUseCase.swift

```
@@ -0,0 +1,70 @@
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import CallKit
+import UserNotifications
+import WireAPI
+import WireDataModel
+import WireLogging
+
+// sourcery: AutoMockable
+protocol GenerateNotificationUseCaseProtocol {
+    func invoke(
+        updateEvents: AsyncStream<[UpdateEvent]>
+    ) async throws -> [UserNotification]
+}
+
+struct GenerateNotificationUseCase: GenerateNotificationUseCaseProtocol {
+
+    let conversationEventBuilder: any ConversationEventNotificationBuilderProtocol
+    let userEventBuilder: any UserEventNotificationBuilderProtocol
+
+    /// Processes the events stream.
+    func invoke(updateEvents: AsyncStream<[UpdateEvent]>) async throws -> [UserNotification] {
+        var notifications = [UserNotification]()
+
+        for await events in updateEvents {
+            for event in events {
+                if let notification = await generateNotification(for: event) {
+                    notifications.append(notification)
+                }
+            }
+        }
+
+        return notifications
+    }
+
+    private func generateNotification(
+        for event: UpdateEvent
+    ) async -> UserNotification? {
+        switch event {
+        case let .conversation(conversationEvent):
+            try? await conversationEventBuilder.buildContent(
+                event: conversationEvent
+            )
+
+        case let .user(userEvent):
+            try? await userEventBuilder.buildContent(
```

todo: log error instead of swallow it

---

remove the throws, this was needed for this one https://github.com/wireapp/wire-ios/pull/2397/commits/122af2773d2acac4a735a1068f06bd0066dccc5b